### PR TITLE
migrate cluster-api 1.6 jobs to community clusters

### PR DESCRIPTION
**Pull request by kranurag7**
this commit updates capi 1.6 jobs to run on eks-prow-build-cluster.

Signed-off-by: Anurag <81210977+kranurag7@users.noreply.github.com>

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-6.yaml [hunk 1]
```diff
@@ -104,16 +106,19 @@ periodics:
...
+            memory: 32Gi
   annotations:
     testgrid-dashboards: sig-cluster-lifecycle-cluster-api-1.6
     testgrid-tab-name: capi-e2e-release-1-6
     testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
     testgrid-num-failures-to-alert: "4"
 - name: periodic-cluster-api-e2e-dualstack-and-ipv6-release-1-6
+  cluster: eks-prow-build-cluster
   interval: 4h
   decorate: true
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-6.yaml:121]: Please also drop the one in l.169

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-6-upgrades.yaml [hunk 2]
```diff
@@ -1,10 +1,9 @@
 periodics:
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-6-upgrades.yaml:1]: Please rebase this PR against the latest master branch, so I can easily diff if we missed something

**Comment by sbueringer** [comment]
Thx!

Alright, given that this PR is ready. Let's change the main jobs on a separate PR

/lgtm
/approve

/hold 
@kranurag7 @furkatgofurov7 Feel free to hold cancel when you're ready

**Review by furkatgofurov7 — COMMENTED** [review]
/approve
/lgtm

**Comment by furkatgofurov7** [comment]
Merging it!

/hold cancel

**Comment by furkatgofurov7** [comment]
Part of https://github.com/kubernetes-sigs/cluster-api/issues/8689

**Pull request merged**